DataConnectors/Gitlab/azuredeploy.json

```diff
@@ -0,0 +1,317 @@
+                "inputs": {
+                    "body": "@{body('User_Audit_Events')}",
+                    "headers": {
+                        "Log-Type": "GitlabUser"
+                    },
+                    "host": {
+                        "connection": {
+                            "name": "@parameters('$connections')['azureloganalyticsdatacollector']['connectionId']"
+                        }
+                    },
+                    "method": "post",
+                    "path": "/api/logs"
+                },
+                "runAfter": {
+                    "User_Audit_Events": [
+                        "Succeeded"
+                    ]
+                },
+                "type": "ApiConnection"
+            },
+            "User_Audit_Events": {
+                "inputs": {
+                    "headers": {
+                        "PRIVATE-TOKEN": "@body('Get_secret')?['value']",
+                        "created_after": "@variables('created_after')"
+                    },
+                    "method": "GET",
+                    "uri": "https://gitlab.com/api/v4/events"
+                },
+                "runAfter": {
+                    "Get_secret": [
+                        "Succeeded"
+                    ]
+                },
+                "type": "Http"
+            }
+        },
+        "contentVersion": "1.0.0.0",
+        "outputs": {},
+        "parameters": {
+            "$connections": {
+                "defaultValue": {},
+                "type": "Object"
+            }
+        },
+        "triggers": {
+            "Recurrence": {
+                "evaluatedRecurrence": {
+                    "frequency": "Minute",
+                    "interval": 5
+                },
+                "recurrence": {
+                    "frequency": "Minute",
+                    "interval": 5
+                },
+                "type": "Recurrence"
+            }
+        }
+    },
+    "parameters": {
+        "$connections": {
+            "value": {
+                "azureloganalyticsdatacollector": {
+                    "connectionId": "[resourceId('Microsoft.Web/connections', variables('azureloganalyticsdatacollector'))]",
+                    "connectionName": "azureloganalyticsdatacollector",
+                    "id": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', resourceGroup().location, '/managedApis/azureloganalyticsdatacollector')]"
+                },
+                "keyvault": {
+                   "connectionId": "[resourceId('Microsoft.Web/connections', variables('AzureKeyVaultConnectionName'))]",
+                    "connectionName": "AzureKeyVaultConnectionName",
+                    "id": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', resourceGroup().location, '/managedApis/keyvault')]"
+                }
+            }
+        }
+    }}}],
+  "outputs": {}
+}
```

Comment:
Should we include the trailing newline like we usually do?

Reply:
> I'm not sure how much the capitalization matters, but I like to follow a brand/[trademark guideline](https://about.gitlab.com/handbook/marketing/corporate-marketing/brand-activation/trademark-guidelines/#also-you-must-comply-with-these-additional-guidelines) if one is published.
> 
> It also feels like we should support multiple project ids or iterate through them if provided a Group ID as @hensonto suggested. Handling a single Group ID is probably fine, but deploying this once per Project ID (repository) does feel like it could be tedious to setup and very easy to miss when setting up a new project.
> 
> I am assuming that we do not currently have a use-case for instance (self-hosted) GitLab at this time. If we do then we would also want to accept the instance URL (defaulted to gitlab.com) and collect from the `/api/v4/audit_events` [API](https://docs.gitlab.com/ee/api/audit_events.html#instance-audit-events) when the instance != gitlab.com (and possibly a `bool` parameter to let the user choose if they can enable this level of audit events)

I have added support for csv argument for group and project ids.